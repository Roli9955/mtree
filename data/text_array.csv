id;val
0;0
1;a
2;a,b
3;a,b,c
4;aaaaa,bbbbb,ccccc,ddddd,eeeee