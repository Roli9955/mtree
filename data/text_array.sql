/* contrib/mtree_gist/data/text_array.sql */

DROP TABLE IF EXISTS mtree_text_array_test CASCADE;

CREATE TABLE mtree_text_array_test (
	id	INTEGER,
	val	mtree_text_array
);

COPY mtree_text_array_test(id, val)
FROM '/home/postgres/test_files/text_array.csv'
DELIMITER ';' CSV HEADER;

DROP INDEX IF EXISTS mtree_text_array_test_idx CASCADE;

CREATE INDEX mtree_text_array_test_idx ON mtree_text_array_test USING gist (
	val gist_mtree_text_array_ops (
		picksplit_strategy	= 'SamplingMinOverlapArea',
		union_strategy		= 'MinMaxDistance'
	)
);

SET enable_seqscan TO OFF;

EXPLAIN ANALYZE
SELECT id, val, (val <-> 'aaaaaaaaaa') AS dst
FROM mtree_text_array_test
ORDER BY (val <-> 'aaaaaaaaaa');

SELECT COUNT(*)
FROM mtree_text_array_test;

SELECT *
FROM mtree_text_array_test;

SELECT id, val, (val <-> 'aaaaaaaaaa') AS dst
FROM mtree_text_array_test
ORDER BY (val <-> 'aaaaaaaaaa')
LIMIT 10;

SELECT *
FROM mtree_text_array_test
WHERE val #<# 'aaaaaaaaaa';

SELECT id, val, (val <-> 'aaaaaaaaaa') AS dst
FROM mtree_text_array_test
ORDER BY (val <-> 'aaaaaaaaaa')
LIMIT 10;
