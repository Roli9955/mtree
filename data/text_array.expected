DROP TABLE
CREATE TABLE
COPY 5
DROP INDEX
CREATE INDEX
SET
 id |              val              
----+-------------------------------
  0 | 0
  1 | a
  2 | a,b
  3 | a,b,c
  4 | aaaaa,bbbbb,ccccc,ddddd,eeeee
(5 rows)

 id |              val              | dst 
----+-------------------------------+-----
  0 | 0                             |   0
  2 | a,b                           |   1
  4 | aaaaa,bbbbb,ccccc,ddddd,eeeee |   1
  3 | a,b,c                         |   1
  1 | a                             |   1
(5 rows)

 id | val 
----+-----
  0 | 0
(1 row)

 id |              val              | dst 
----+-------------------------------+-----
  0 | 0                             |   0
  2 | a,b                           |   1
  4 | aaaaa,bbbbb,ccccc,ddddd,eeeee |   1
(3 rows)

